### PR TITLE
Get email property commands

### PR DESCRIPTION
**Pull request by openbots-su**

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        [SampleUsage("")]
+        [Remarks("")]
+        public string v_Property { get; set; }
+
+        [Required]
+        [DisplayName("Output MimeMessage Property Variable")]
+        [Description("Create a new variable or select a variable from the list.")]
+        [SampleUsage("{vUserVariable}")]
+        [Remarks("Variables not pre-defined in the Variable Manager will be automatically generated at runtime.")]
+        [Editor("ShowVariableHelper", typeof(UIAdditionalHelperType))]
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs:72]: Please remove the variable helper from the output

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                    break;
+            }
+        }
+
+        public override List<Control> Render(IfrmCommandEditor editor, ICommandControls commandControls)
+        {
+            base.Render(editor, commandControls);
+
+            RenderedControls.AddRange(commandControls.CreateDefaultInputGroupFor("v_MimeMessage", this, editor));
+            RenderedControls.AddRange(commandControls.CreateDefaultInputGroupFor("v_Property", this, editor));
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs:188]: This should be a dropdowngroup

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            RenderedControls.AddRange(commandControls.CreateDefaultOutputGroupFor("v_OutputUserVariableName", this, editor));
+
+            return RenderedControls;
+        }
+
+        public override string GetDisplayValue()
+        {
+            return base.GetDisplayValue() + $" [From '{v_MimeMessage}' - Get '{v_Property} value' - Store property in '{v_OutputUserVariableName}']";
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs:196]: please use APA title capitalization convention https://capitalizemytitle.com/

File: OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailPropertyCommand.cs [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+                    break;
+                case "UnRead":
+                    v_MailItem.UnRead.StoreInUserVariable(engine, v_OutputUserVariableName);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+}
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailPropertyCommand.cs:314]: missing Render and DisplayValue code

**Review by openbots-ff — CHANGES_REQUESTED** [review]
Please update the UIImage.cs file as well.

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs (hunk 1 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs]: Addressed in 0ff24ed9f317606cece239b2c1153d09e916d8f4

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs (hunk 2 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs]: Addressed in 0ff24ed9f317606cece239b2c1153d09e916d8f4

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs (hunk 3 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessageProperty.cs]: Addressed in 0ff24ed9f317606cece239b2c1153d09e916d8f4

File: OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailPropertyCommand.cs (hunk 4 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailPropertyCommand.cs:330]: Implemented in 17285f3a7a725e856268dc2ba9ef74dccb0b322c

File: OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailPropertyCommand.cs [hunk 5]
```diff
@@ -310,5 +310,21 @@ public override void RunCommand(object sender)
...
+            RenderedControls.AddRange(commandControls.CreateDefaultInputGroupFor("v_MailItem", this, editor));
+            RenderedControls.AddRange(commandControls.CreateDefaultDropdownGroupFor("v_Property", this, editor));
+            RenderedControls.AddRange(commandControls.CreateDefaultOutputGroupFor("v_OutputUserVariableName", this, editor));
+
+            return RenderedControls;
+        }
+
+        public override string GetDisplayValue()
+        {
+            return base.GetDisplayValue() + $" [From '{v_MailItem}' - Get '{v_Property} Value' - Store Property in '{v_OutputUserVariableName}']";
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailPropertyCommand.cs:327]:
```suggestion
            return base.GetDisplayValue() + $" [From '{v_MailItem}' - Get '{v_Property}' Value - Store Property in '{v_OutputUserVariableName}']";
```

File: OpenBots.Studio/UI/CustomControls/CustomUIControls/UIImage.cs [hunk 6]
```diff
@@ -214,6 +214,8 @@ public static class UIImage
             uiImages.Add("ExtendQueueItemCommand", Resources.command_queueitem);
             uiImages.Add("SetQueueItemStatusCommand", Resources.command_queueitem);
             uiImages.Add("WorkQueueItemCommand", Resources.command_queueitem);
+            uiImages.Add("GetMimeMessagePropertyCommand", Resources.command_queueitem);
```
Comment (openbots-ff) [inline OpenBots.Studio/UI/CustomControls/CustomUIControls/UIImage.cs:217]: please use the appropriate Resource. These aren't queueitem commands

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessagePropertyCommand.cs [hunk 7]
```diff
@@ -185,15 +251,15 @@ public override List<Control> Render(IfrmCommandEditor editor, ICommandControls
...
+            RenderedControls.AddRange(commandControls.CreateDefaultDropdownGroupFor("v_Property", this, editor));
             RenderedControls.AddRange(commandControls.CreateDefaultOutputGroupFor("v_OutputUserVariableName", this, editor));
 
             return RenderedControls;
         }
 
         public override string GetDisplayValue()
         {
-            return base.GetDisplayValue() + $" [From '{v_MimeMessage}' - Get '{v_Property} value' - Store property in '{v_OutputUserVariableName}']";
+            return base.GetDisplayValue() + $" [From '{v_MimeMessage}' - Get '{v_Property} Value' - Store Property in '{v_OutputUserVariableName}']";
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessagePropertyCommand.cs:262]:
```suggestion
            return base.GetDisplayValue() + $" [From '{v_MimeMessage}' - Get '{v_Property}' Value - Store Property in '{v_OutputUserVariableName}']";
```

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailPropertyCommand.cs [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+                    item.TaskSubject.StoreInUserVariable(engine, v_OutputUserVariableName);
+                    break;
+                case "To":
+                    item.To.StoreInUserVariable(engine, v_OutputUserVariableName);
+                    break;
+                case "UnRead":
+                    item.UnRead.ToString().StoreInUserVariable(engine, v_OutputUserVariableName);
+                    break;
+                default:
+                    break;
```
Comment (openbots-ch) [inline OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailPropertyCommand.cs:332]: See previous comment.

**Review by openbots-ch — CHANGES_REQUESTED** [review]

File: OpenBots.Studio/UI/CustomControls/CustomUIControls/UIImage.cs (hunk 6 above)
Comment (openbots-su) [reply OpenBots.Studio/UI/CustomControls/CustomUIControls/UIImage.cs]: Corrected in b0d30b8015b87359003a8592213e4e7b45307b95

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessagePropertyCommand.cs (hunk 7 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessagePropertyCommand.cs]: Addressed in 6a35264215e279a3c4171c42587d26c8b9ba77e3

File: OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessagePropertyCommand.cs [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+                    {
+                        output = output + item.ToString() + ";";
+                    }
+                    output.Substring(0, output.Length - 1).StoreInUserVariable(engine, v_OutputUserVariableName);
+                    break;
+                case "XPriority":
+                    email.XPriority.ToString().StoreInUserVariable(engine, v_OutputUserVariableName);
+                    break;
+                default:
+                    break;
```
[the remark this answers is not in the document]
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Email/GetMimeMessagePropertyCommand.cs]: Addressed in 6a35264215e279a3c4171c42587d26c8b9ba77e3

**Review by openbots-ch — DISMISSED** [review]

File: OpenBots.Commands/OpenBots.Commands.Loop/OpenBots.Commands.Loop.csproj [hunk 10]
```diff
@@ -36,16 +36,54 @@
...
     </Reference>
     <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Newtonsoft.Json.12.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
+      <EmbedInteropTypes>False</EmbedInteropTypes>
     </Reference>
     <Reference Include="Serilog, Version=2.0.0.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Serilog.2.10.0\lib\net46\Serilog.dll</HintPath>
+      <EmbedInteropTypes>False</EmbedInteropTypes>
+    </Reference>
+    <Reference Include="System">
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Loop/OpenBots.Commands.Loop.csproj:52]: please undo everything in this file except for the having set outlook interop to false

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.Loop/OpenBots.Commands.Loop.csproj (hunk 10 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Loop/OpenBots.Commands.Loop.csproj]: Addressed in 6055e294a1565547ef8d15716d8039a799d13daf

**Review by openbots-ff — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]

**Pull request merged**